### ISSUE TITLE
no pushToken object in ids listener android

### ISSUE THREAD

**Issue by jigaryadav** [OneSignal/react-native-onesignal#500]
**Description:**
after updating package to 3.2.1 i am not getting pushToken, even after i downgraded my package but  not getting any token at all I have changed my config according to package version 

**Environment**
current version "react-native-onesignal": "3.2.1"

**Comment by Nightsd01** [comment]
#503 is a duplicate of this issue, but since 503 has a lot more detail I will be closing this one.

**Comment by yaseralimardany** [comment]
how you can solve this issue?

**Comment by jigaryadav** [comment]
@yaseralimardany  i just downgraded my version, latest version has some issues so will upgrade later

**Comment by yaseralimardany** [comment]
downgrade to which version?

**Comment by Nightsd01** [comment]
>downgrade to which version? ...

@yaseralimardany he’s talking about 3.1.4. You can downgrade by simply changing the version to 3.1.4 in your `package.json` and running `npm install`...

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by Nightsd01** [comment]
We resolved a similar issue to this but that issue only effected iOS, so this must be a separate issue. 

@jigaryadav  when you downgraded to 3.1.4 did that resolve the issue? And if you update to 3.2.3 does the issue come back again?

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by yaseralimardany** [comment]
Yes, this issue was appeared in android version. I tried 3.2.2 and 3.2.3 version.
I think its related to not add compile "com.google.android.gms:play-services-base" in the dependencies. After I add that, I had compile error but push token was not null. By deleting that, pushToken will be null.
I think in newer version of onesignal that not need to add play-service-gcm or base, pushToken will be null.

**Comment by yaseralimardany** [comment]
By adding these dependencies, app will be crash and pushToken was not null.

app crash error:
Process: com.motoshubapp, PID: 26653
                                                                 java.lang.IllegalStateException: Default FirebaseApp is not initialized in this process com.motoshubapp. Make sure to call FirebaseApp.initializeApp(Context) first.

**Comment by jigaryadav** [comment]
@Nightsd01 I downgraded to 3.1.4  and its working fine for me

**Comment by yaseralimardany** [comment]
I am in 3.1.4 but app crashed. when i add compile("com.google.android.gms:play-services-gcm:+") app crashed but pushToken was not null.
app crashed because of firebaseApp initialize.

**Comment by adirzoari** [comment]
@yaseralimardany same issue when I have firebase and onesignal. push token is null in android. did you find solution? I have `    "react-native-onesignal": "^3.2.3",
`

**Comment by yaseralimardany** [comment]
this is my root build.gradle

// Top-level build file where you can add configuration options common to all sub-projects/modules.

```
buildscript {
    repositories {
        jcenter()
        google()
        mavenLocal()
    }
    dependencies {
        classpath 'com.android.tools.build:gradle:3.0.1'

        // NOTE: Do not place your application dependencies here; they belong
        // in the individual module build.gradle files
    }
}

allprojects {
    repositories {
         configurations.all {
            resolutionStrategy.eachDependency { DependencyResolveDetails details ->
                def requested = details.requested
                if (requested.group == 'com.google.android.gms') {
                    details.useVersion '12.0.1'
                }
            }
        }
        mavenLocal()
        jcenter()
        google()
        maven {
            url "https://jitpack.io"
        }
        maven {
            url 'https://maven.google.com'
        }
        maven {
            // All of React Native (JS, Obj-C sources, Android binaries) is installed from npm
            url "$rootDir/../node_modules/react-native/android"
        }
    }
}

subprojects {
    afterEvaluate {project ->
        if (project.hasProperty("android")) {
            android {
                compileSdkVersion 26
                buildToolsVersion '26.0.2'
            }
        }
    }
}
```

in app build.gradle
```
compileSdkVersion 27
buildToolsVersion "27.0.3"
compile "com.android.support:appcompat-v7:27.1.0"
```

compile, run, load and subscribe on OneSignal after OneSignal.init

**Comment by adirzoari** [comment]
@yaseralimardany onesignal works for you and you get notification in android and ios?

**Comment by yaseralimardany** [comment]
I dont check ios yet but everythings in android is OK. Its important to know that, eventListerens of onesignal does not works with reload function in debug mode.

**Comment by Nightsd01** [comment]
We believe this issue has been addressed in the new update (3.2.4). Anyone who experienced this issue, please let me know if the issue still occurs and I'll be happy to reopen & investigate further
